tests/subspace_search/test_mist.py

```diff
@@ -0,0 +1,202 @@
+    )
+
+    evaluation_result = subspace_evaluation.subspace_fit(
+        np.linalg.svd(data.A, full_matrices=False)[0], estimation_result.A_1.numpy()
+    )
+
+    random_matrices_results = [
+        subspace_evaluation.subspace_fit(
+            np.linalg.svd(data.A, full_matrices=False)[0],
+            np.linalg.svd(rng.random(size=data.A.shape), full_matrices=False)[0],
+        )["Value"]["Difference between subspaces"]
+        for _ in range(1000)
+    ]
+
+    assert manifold_utils.is_orthonormal(estimation_result.A_1, atol=1e-2)
+
+    assert evaluation_result["Value"]["Difference between subspaces"] < np.quantile(random_matrices_results, 0.05)
+
+
+transforms = [
+    [lambda x: np.sum(x**3, axis=1) + rng.normal(0, 0.05, size=(x.shape[0])).astype(np.float32)],
+]
+
+
+@pytest.mark.parametrize(
+    "N",
+    [8000],
+)
+@pytest.mark.parametrize(
+    "n",
+    [8, 16],
+)
+@pytest.mark.parametrize(
+    "d",
+    [2, 3, 4],
+)
+@pytest.mark.parametrize(
+    "transform",
+    transforms,
+)
+def test_nonlinear_data(N: int, n: int, d: int, transform: List[Callable[[np.ndarray], np.ndarray]]):
+
+    data = synthetic.generate(N, n, d, fs=transform)
+
+    data.X = StandardScaler().fit_transform(data.X)
+    data.Z = StandardScaler().fit_transform(data.Z)
+
+    estimation_result = mist_estimation.fit(
+        data.X,
+        data.Z,
+        model_kwargs=dict(x_size=n, z_max_size=len(transform), subspace_size=d),
+        trainer_kwargs=dict(max_epochs=512, gpus=0),
+    )
+
+    evaluation_result = subspace_evaluation.subspace_fit(data.A, estimation_result.A_1.numpy())
+
+    random_matrices_results = [
+        subspace_evaluation.subspace_fit(
+            np.linalg.svd(data.A, full_matrices=False)[0],
+            np.linalg.svd(rng.random(size=data.A.shape), full_matrices=False)[0],
+        )["Value"]["Difference between subspaces"]
+        for _ in range(1000)
+    ]
+
+    assert manifold_utils.is_orthonormal(estimation_result.A_1, atol=1e-2)
+
+    assert evaluation_result["Value"]["Difference between subspaces"] < np.quantile(random_matrices_results, 0.05)
+
+
+@pytest.mark.parametrize(
+    "N",
+    [32000],
+)
+@pytest.mark.parametrize(
+    "n",
+    [3],
+)
+@pytest.mark.parametrize(
+    "d",
+    [1],
+)
+def test_probabilistic_pca_data_mi_estimation_full_space(N: int, n: int, d: int) -> None:
+
+    data = ppca.generate(N, n, d, d)
+    true_mi = ppca.generative_model_mutual_information(data.A, data.sigma)
+
+    data.X = StandardScaler().fit_transform(data.X)
+
+    estimation_result = mist_estimation.fit(
+        data.X,
+        data.Z,
+        model_kwargs=dict(x_size=n, z_max_size=d, subspace_size=(n, d), mine_network_width=128),
+        trainer_kwargs=dict(max_epochs=512, gpus=0),
+    )
+
+    assert manifold_utils.is_orthonormal(estimation_result.A_1, atol=1e-2)
+
+    assert estimation_result.mutual_information == pytest.approx(true_mi, 0.1)
+
+
+@pytest.mark.parametrize(
+    "N",
+    [32000],
```

Comment:
(Same, also the number of epochs is quite high. We don't have so much capacity on the servers, unfortunately :slightly_frowning_face: )

Reply:
Oh nooo, sorry, I didn't know this! Will change this then!